src/app/clusters/fault-injection-server/fault-injection-server.cpp

```diff
@@ -93,4 +93,66 @@ bool emberAfFaultInjectionClusterFailAtFaultCallback(CommandHandler * commandObj
     return true;
 }
 
+bool emberAfFaultInjectionClusterFailRandomlyAtFaultCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                             const Commands::FailRandomlyAtFault::DecodableType & commandData)
+{
+    if (commandPath.mClusterId != Clusters::FaultInjection::Id)
+    {
+        // We shouldn't have been called at all.
+        commandObj->AddStatus(commandPath, Status::UnsupportedCluster);
+        return true;
+    }
+
+    if (commandData.percentage > 100)
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+#if CHIP_WITH_NLFAULTINJECTION
+    Status returnStatus                             = Status::Success;
+    nl::FaultInjection::Manager * faultInjectionMgr = nullptr;
+
+    switch (commandData.type)
+    {
+    case FaultType::kSystemFault:
+        faultInjectionMgr = &chip::System::FaultInjection::GetManager();
+        break;
+    case FaultType::kInetFault:
+        faultInjectionMgr = &chip::Inet::FaultInjection::GetManager();
+        break;
+    case FaultType::kChipFault:
+        faultInjectionMgr = &chip::FaultInjection::GetManager();
+        break;
+    default:
+        ChipLogError(Zcl, "FaultInjection: Unsupported Fault type received");
+        returnStatus = Status::InvalidCommand;
```

Comment:
status is also set as invalid below if faultInjectionMgr is not set to non-null. Should we also set it here? Do we need both logs at line 128 and 147?

Reply:
Acked